packages/fxa-auth-server/lib/oauth/jwt_access_token.js

```diff
@@ -18,7 +19,11 @@ exports.create = async function generateJWTAccessToken(accessToken, grant) {
   // always include the client_id in the `aud` claim. A future iteration of this code
   // should instead infer an appropriate default `aud` based on the requested scopes.
   // Ref https://github.com/mozilla/fxa/issues/4962
-  const audience = grant.resource ? [clientId, grant.resource] : clientId;
+  const audience = grant.resource
+    ? [clientId, grant.resource]
+    : grant.scope.contains(OAUTH_SCOPE_OLD_SYNC)
+    ? 'https://token.services.mozilla.com'
```

Comment:
nit: this could profitably be extracted into a constant.

Better yet, maybe we can get it from the same place that configures the value advertised in https://accounts.firefox.com/.well-known/fxa-client-configuration? I think it would be valuable to have the value here match the one advertised to clients there, for consistency.